### DISCUSSION TITLE
Source formatting

### DISCUSSION THREAD

**Discussion by chriswilty** [giraud/reasonml-idea-plugin#396] [Feature request]
Looks like the plugin is using rescript to do formatting (ResFormatProcess.java) which is fully opinionated about format and doesn't allow any user opinions about trailing commas, whitespace, indentation type or size, etc.

If a user has editorconfig set up, then I would assume that should be applied after rescript formatting, so should allow control over a few lightweight aspects of formatting such as indentation. However, that is not currently working - it seems editorconfig is being ignored for .res/.resi files. Is that a bug?

Maybe a better option for users who really care about formatting beyond editorconfig would be to try implementing a prettier plugin, similar to https://www.npmjs.com/package/prettier-plugin-reasonml. I might give that a go anyway, just for kicks - looks quite trivial.

**Comment by giraud** [comment]
This is intentional. There is only one formatting and it is done by the rescript binary.

I have absolutely no plan to implement a custom formatter in the plugin. so yes, an editorconfig file is useless.
It might be better to create a separate pluging for people who want custom formatting.

For info, you can disable 'format on save' in the settings 
![image](https://user-images.githubusercontent.com/239612/206686803-665820d2-9967-41ed-a2e3-103d90dc6e2c.png)
